cdk/examples/generative_ai_service/web-app/pages/2_llm_langchain.py

```diff
@@ -46,17 +47,41 @@ def create_chunks(context):
 
     return docs
 
-def invoke_map_reduce(context, query, endpoint_name, region_name):
+def simple_map_reduce(context, map_query, reduce_query, endpoint_name, region_name):
+    docs = create_chunks(context)
+    content_handler = ContentHandler()
+    llm = SagemakerEndpoint(
+                    endpoint_name = endpoint_name,
+                    region_name= region_name,
+                    content_handler=content_handler
+                )
+
+    map_template = """In the following text find all answers to the question which is delimited by ```. {text}. Provide answer as complete sentence.""" + """```""" + map_query + """```\n"""
+    map_prompt = PromptTemplate(template=map_template,
+                                input_variables=["text"])
+    map_chain = LLMChain(llm=llm,prompt=map_prompt)
+
+    output_list =[]
+    for doc in docs:
+        output = map_chain.run(doc)
+        output_list.append(output)
+
+    reduce_template = """In the following comma separate text list find all answers to the question which is delimited by ```. {text}. Provide answer as complete sentence.""" + """```""" + reduce_query + """```\n"""
```

Comment:
```suggestion
    reduce_template = """In the following comma separated text list, find all answers to the question which is delimited by ```. {text}. Provide answer as complete sentence.""" + """```""" + reduce_query + """```\n"""
```